### PR TITLE
Refer to case where Mellanox adapter used by VM NIC.

### PR DESCRIPTION
**Pull request by yossisegev**

### CODE REVIEW

**Comment by yossisegev** [comment]
@phoracek @danken@redhat.com @germano@redhat.com fbaudin@redhat.com
I would appreciate your review here.
Thanks you!

File: templates/testpmd/README.md [hunk 1]
```diff
@@ -95,9 +95,11 @@ Note that `containerDisk` value should be replaced with the actual repo where th
 
 
 ## Extra steps on the running VM
-DPDK requires that the driver bound to the interface that uses it (the SR-IOV interface in the VM) is vfio-pci.
+DPDK requires that the driver bound to the interface that uses it (the SR-IOV interface in the VM) is one of few models.
```
Comment (ormergi) [inline templates/testpmd/README.md:98]: Is'nt DPDK requiters the VF's to be bound to the `vfio-pci` driver?

File: templates/testpmd/README.md [hunk 2]
```diff
@@ -95,9 +95,11 @@ Note that `containerDisk` value should be replaced with the actual repo where th
...
+If this driver model is iavf, then the VF interface (i.e. the SR-IOV NIC in the VM) should be actively bound to an vfio-pci driver by the user.
```
Comment (ormergi) [inline templates/testpmd/README.md:99]: I would omit "model" it may confuse the reader since this sentence refers to the VF driver, and rephrase to something like:
If the VF interface driver is `iavf` it should be changed to `vfio-pci` manually.

File: templates/testpmd/README.md [hunk 3]
```diff
@@ -95,9 +95,11 @@ Note that `containerDisk` value should be replaced with the actual repo where th
...
 It might be, however, that the driver supported by the node is different. In that case, the user would have to actively replace the default driver (e.g. iavf) which is bound to the SR-IOV NIC in the VM.
-You can achieve that by running the scripts/change-nic-driver.sh script in the VM (copy its contents to the VM).
+This can be achieved by running the scripts/change-nic-driver.sh script in the VM (copy its contents to the VM).
+If the driver in use is a Mellanox driver - the script will exit, and the user should refer to [https://doc.dpdk.org/guides/nics/mlx5.html#usage-example] (https://doc.dpdk.org/guides/nics/mlx5.html#usage-example).
```
Comment (ormergi) [inline templates/testpmd/README.md:102]: As far as I know changing the driver for Mellanox card VF's is the same as for Intel cards.
We used to have some nodes with Intel cards and some with Mellanox card on Kubevirt U/S CI cluster,
the logic for changing the driver was the same, [here is how it is done on kubevirtci](https://github.com/kubevirt/kubevirtci/blob/main/cluster-up/cluster/kind-1.22-sriov/sriov-node/configure_vfs.sh#L5-L30).

File: templates/testpmd/scripts/change-nic-driver.sh [hunk 4]
```diff
@@ -11,6 +11,14 @@ if [ "$driver" == "vfio-pci" ]
...
   echo "Driver is vfio-pci, no extra action needed."
   exit
+elif [[ "$driver" == *"mlx"* ]]
+then
+  echo "Driver is a Mellanox driver, please refer to https://doc.dpdk.org/guides/nics/mlx5.html#usage-example in order to replace it."
+  exit
+elif [ "$driver" != "iavf" ]
+then
+  echo "Driver is $driver, and cannot be handled here."
+  exit
```
Comment (ormergi) [inline templates/testpmd/scripts/change-nic-driver.sh:21]: Why the script exist if the driver is not `iavf`? I think it should change it either way.

File: templates/testpmd/scripts/change-nic-driver.sh [hunk 5]
```diff
@@ -32,7 +40,7 @@ modprobe vfio-pci
 
 # Enable unsafe IOMMU.
 echo 1 > /sys/module/vfio/parameters/enable_unsafe_noiommu_mode
-conf
+echo "options vfio enable_unsafe_noiommu_mode=1" > /etc/modprobe.d/vfio-noiommu.conf
```
Comment (ormergi) [inline templates/testpmd/scripts/change-nic-driver.sh:43]: I think this change is not related to this PR, also I saw the same change on https://github.com/RHsyseng/cnv-supplemental-templates/pull/7/files#diff-9a9eb2317c6991aa55853383742de3b38bcdd554c63b0aa63ff9669963146771R35

**Review by ormergi — COMMENTED** [review]
Hi Yossi, 
With this PR changes the README suggests to copy the script `change-nic-driver.sh` to inside the VM manually which may be cumbersome.
I think its better if we guide the users to add the script to the VM image instead, WDYT?

**Comment by ormergi** [comment]
@yossisegev ping

**Comment by yossisegev** [comment]
Abandoned because everything which is still relevant in this PR is covered in https://github.com/RHsyseng/cnv-supplemental-templates/pull/7
